### PR TITLE
Bigupload

### LINKED ISSUE #55: Rows method reads first N rows but then continues loading the data
When I call `file.rows({size:100})` it returns me the first 100 rows, however, if I check my browser's network, it is still loading the data.

### LINKED ISSUE #89: [upload] browser crashes while trying to upload large file (~600Mb)
Browser crashes while trying to upload a large CSV file (~600Mb) for the resource

## Acceptance

- [x] Generate hash MD5 and SHA256 for large files 

## Tasks

- [x] slipt the file in chunks and generate hash chunk by chunk and after compile in the final hash

## Analysis

There are two options:
1. Uploading streams - **Problem**: stream upload is not supported by old browsers, since we should use [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) 
**[Udpates]:** This is not applicable. We should use FormData() and the file stream appened on it which doens't meet the way it suppoused to be done for upload file to the storage. Currently we pass a buffer, not a FormData object having a bufer/file/stream, this won't be handled by the storage service.

2. Multipart - **Problem**:  We can directly modify queries etc of upload storage request, however this depends on which storage we use, for each provider we have to make custom changes, which will basically endup having our own Uppy plugin.

### PR DESCRIPTION
**Pull request by risenW**
Task

- [x] Fix #89 [upload] browser crashes while trying to upload large file (~600Mb)
- [x] Fix #55 Rows method reads first N rows but then continues loading the data

### CODE REVIEW

File: datajs/src/parser/csv.js [hunk 1]
```diff
@@ -132,3 +129,44 @@ export class Uint8ArrayToStringsTransformer {
     }
   }
 }
+
```
Comment (mariorodeghiero) [inline datajs/src/parser/csv.js]: Maybe could be good to add a JS DOC for the function.

**Review by mariorodeghiero — APPROVED** [review]
LGTM

File: datajs/src/parser/csv.js (hunk 1 above)
Comment (risenW) [reply datajs/src/parser/csv.js]: Great idea, will do that!

File: datajs/src/parser/csv.js [hunk 2]
```diff
@@ -25,10 +25,9 @@ export async function guessParseOptions(file) {
   if (file.displayName === 'FileLocal') {
     const stream = await file.stream({ end: 50000 })
     text = await toString(stream)
-
   } else if (file.displayName === 'FileInterface') {
-    text = await file.descriptor.text()
-
+    // text = await file.descriptor.text()
+    text = await getChunkText(file)
```
Comment (kmanaseryan) [inline datajs/src/parser/csv.js]: @risenW Nice replacement  👍 Can we delete the commented function?

File: datajs/src/parser/csv.js (hunk 1 above)
Comment (kmanaseryan) [reply datajs/src/parser/csv.js]: Yea, let's always use JSDoc👍

File: datajs/src/parser/csv.js [hunk 3]
```diff
@@ -132,3 +129,44 @@ export class Uint8ArrayToStringsTransformer {
     }
   }
 }
+
+async function getChunkText(file, size = 10) {
```
Comment (kmanaseryan) [inline datajs/src/parser/csv.js]: @mariorodeghiero has implemented similar function called [readChunked()](https://github.com/datopian/data.js/blob/master/datajs/src/browser-utils/utils.js#L51). Can we merge those two? Also I would keep this in utils.

**Review by kmanaseryan — CHANGES_REQUESTED** [review]

File: datajs/src/parser/csv.js (hunk 2 above)
Comment (risenW) [reply datajs/src/parser/csv.js]: Sure

File: datajs/src/parser/csv.js (hunk 3 above)
Comment (risenW) [reply datajs/src/parser/csv.js]: Okay, will work on that now.

File: datajs/src/browser-utils/utils.js [hunk 4]
```diff
@@ -3,16 +3,19 @@ import { Readable } from 'stream'
...
  * Return node like stream so that parsers work.
  * Transform browser's Reader to string, then create a nodejs stream from it
- * @param {object} reader
- * @param {number} size
+ * @param {object} reader A file stream reader from the browser input
+ * @param {number} size Size of data to return
+ * @param {boolean} return_chunk whether to return a chunk in string format or a node stream
  */
-export async function toNodeStream(reader, size) {
+export async function toNodeStream(reader, size, return_chunk = false) {
```
Comment (kmanaseryan) [inline datajs/src/browser-utils/utils.js]: `returnChunk`: use camel case

File: datajs/src/browser-utils/utils.js [hunk 5]
```diff
@@ -26,29 +29,40 @@ export async function toNodeStream(reader, size) {
...
+
   return nodeStream
 }
 
 export function isFileFromBrowser(file) {
   return file instanceof File
 }
 
-export function readChunked(file, chunkCallback, endCallback){
+export function readChunked(file, chunkCallback, endCallback) {
```
Comment (kmanaseryan) [inline datajs/src/browser-utils/utils.js]: OK, we have missed JSDoc here previously, so add them to make the code improved. Also just to have better naming let's make the following changes:

`chunkCallback` -> `next`
`endCallback` -> `done`
`readChunked` -> `readChunk`

**Review by kmanaseryan — CHANGES_REQUESTED** [review]

File: datajs/src/browser-utils/utils.js (hunk 4 above)
Comment (risenW) [reply datajs/src/browser-utils/utils.js]: True, thanks

File: datajs/src/browser-utils/utils.js (hunk 5 above)
Comment (risenW) [reply datajs/src/browser-utils/utils.js]: Okay, on it

**Review by kmanaseryan — APPROVED** [review]
👍

**Pull request merged**